### PR TITLE
Added some simple (ANSI.SYS compatible) cursor movements:

### PR DESCRIPTION
**Pull request by nahkampf**
- CUU - Up
- CUD - Down
- CUF - Forward
- CUB - Back
- CUP - Move to position (x,y)

### CODE REVIEW

File: src/Traits/EscapeSequences/CUP.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Erase the entire current line.
+     * @param  boolean $outputNow Echo the character right now, or add it to the buffer building?
+     * @return Ansi    self, for chaining
+     */
+    public function cursorPosition($coords = [1,1])
+    {
+        return $this->cuu($coords);
```
Comment (bramus) [inline src/Traits/EscapeSequences/CUP.php]: This should be `$this->cup`, no?

File: src/Traits/EscapeSequences/CUU.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        $this->writer->write(
+            new \Bramus\Ansi\ControlSequences\EscapeSequences\CUU($data)
+        );
+
+        // Afford chaining
+        return $this;
+    }
+
+    /**
+     * Erase the entire current line.
```
Comment (bramus) [inline src/Traits/EscapeSequences/CUU.php]: This comment is not up-to-date

File: src/Traits/EscapeSequences/CUP.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        $this->writer->write(
+            new \Bramus\Ansi\ControlSequences\EscapeSequences\CUP($data)
+        );
+
+        // Afford chaining
+        return $this;
+    }
+
+    /**
+     * Erase the entire current line.
```
Comment (bramus) [inline src/Traits/EscapeSequences/CUP.php]: This comment is not up-to-date

File: src/Traits/EscapeSequences/CUP.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+namespace Bramus\Ansi\Traits\EscapeSequences;
+
+/**
+ * Trait containing the CUP Escape Function Shorthands
+ */
+trait CUP
+{
+    /**
+     * Manually use CUP (Cursor position)
+     * @param  array $parameterByte Parameter byte to the CUP Escape Code
```
Comment (bramus) [inline src/Traits/EscapeSequences/CUP.php]: This comment is not up-to-date

File: src/ControlSequences/EscapeSequences/CUU.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Bramus\Ansi\ControlSequences\EscapeSequences;
+
+class CUU extends Base
+{
+    // This EscapeSequence has ParameterByte(s)
+    use \Bramus\Ansi\ControlSequences\Traits\HasParameterBytes;
+
+    /**
+     * SGR - SELECT GRAPHIC RENDITION
```
Comment (bramus) [inline src/ControlSequences/EscapeSequences/CUU.php]: This comment is not up-to-date

**Review by bramus — CHANGES_REQUESTED** [review]
Hi @nahkampf, thanks for these additions.

Could you tweak the PR more, and

- [ ] Provide update PHPDoc comments? They don't match the contents of the functions _(I've marked only a few)_
- [ ] Fix the wrong call to `cuu` (which should be `cup`)

Thanks.

**Comment by nahkampf** [comment]
Sorry @bramus , that was a sloppy PR, has been fixed now!

**Review by bramus — APPROVED** [review]

**Comment by bramus** [comment]
Thanks for updating the PR this fast, @nahkampf.

Will take a look some time next week to merge and provide an updated README + extra tests.

**Comment by bramus** [comment]
I've reworked/fixed a few things:

- Reshuffle some code parts
- Rename arguments
- Fix comments
- Fix default values

And also made these additions:

- Added tests
- Updated README

Changes can be found at https://github.com/bramus/ansi-php/tree/feature-cursor. What do you think?

While I was at it, I create a small demo that leverages CUB (and EL): a little ascii-spinner:

php-ascii-spinner

The code for the spinner is included in the README.

**Comment by nahkampf** [comment]
> I've reworked/fixed a few things:

That's pretty hot! I'm thinking of implementing cursor movements and save/restore screen to build fixed width/length input fields for a thing, I'll let you know if I get it working like I want it and I could post a gist or something. 👍

**Comment by bramus** [comment]
> I'm thinking of implementing cursor movements and save/restore screen to build fixed width/length input fields for a thing, I'll let you know if I get it working like I want it and I could post a gist or something. 👍

Cool. I'll go ahead and merge my adjusted branch so that you can build further upon it.

**Pull request merged**